### PR TITLE
[CP] Add report feature on citizen project

### PR DESCRIPTION
**Pull request by B-Galati**

### CODE REVIEW

**Comment by B-Galati** [comment]
Sorry, this PR is too big

File: app/config/security.yml [hunk 1]
```diff
@@ -172,6 +172,12 @@ security:
         ROLE_ADMIN_BOARD_MEMBER_ROLES:
             - ROLE_APP_ADMIN_ROLE_ALL
 
+        ROLE_ADMIN_REPORTS:
+            - ROLE_APP_ADMIN_REPORT_APPROVE
+            - ROLE_APP_ADMIN_REPORT_VIEW
+            - ROLE_APP_ADMIN_REPORT_LIST
+            - ROLE_APP_ADMIN_REPORT_ALL
```
Comment (Remg) [inline app/config/security.yml]: `ROLE_APP_ADMIN_REPORT_ALL` implicetely includes  `LIST` and `VIEW` roles.

Comment (B-Galati) [reply app/config/security.yml]: Thank you I'll remove them ;-)

File: app/migrations/Version20171211123655.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
```
Comment (thomasbisignani) [inline app/migrations/Version20171211123655.php]: This line is usefull

File: templates/admin/report/_author.html.twig [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        {{ author.fullName }}
+    </a>
+    <br />
+    {{ author.emailAddress }}
+    {% if author.phone %}
+        <br />
+        {{ author.phone|phone_number_format }}
+    {% endif %}
+{% else %}
+    {{ author.fullName }}
```
Comment (Remg) [inline templates/admin/report/_author.html.twig]: :thinking:

File: templates/citizen_project/_layout.html.twig [hunk 4]
```diff
@@ -34,7 +34,10 @@
                                 <p>En tant que porteur de projet citoyen, vous ne pouvez pas cesser de le suivre.</p>
                             {% endif %}
 
-                            <a href="#" class="link--blue--dark link--no-decor">Signaler un abus</a>
+                            {% if is_granted('REPORT') %}
+                                <a href="{{ path('app_report', {type:'citizen-project', uuid:citizen_project.uuid, redirectUrl: app.request.pathInfo}) }}" class="report-link link--blue--dark link--no-decor">Signaler un abus</a>
```
Comment (Remg) [inline templates/citizen_project/_layout.html.twig]: missing spaces: `type: 'citizen-project', uuid: citizen_project.uuid`

File: templates/report/report.html.twig [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+
+                    <div class="form l__col">
+                        <div>
+                            {{ form_row(form.reasons) }}
+                        </div>
+
+                        <div>
+                            {{ form_errors(form.comment) }}
+                            {{ form_widget(form.comment, { attr: {'class': 'form__textbox'} }) }}
```
Comment (Remg) [inline templates/report/report.html.twig]: `{class: 'form__textbox'}`

File: templates/adherent/terminate_membership.html.twig [hunk 6]
```diff
@@ -76,7 +76,7 @@
                     {% endfor %}
                     <div id="unregistration_another_reason_comment">
                         {{ form_errors(form.comment) }}
-                        {{ form_widget(form.comment, { attr: {'class': 'form__textbox'} }) }}
+                        {{ form_widget(form.comment, { attr: { class: 'form__textbox'} }) }}
```
Comment (Remg) [inline templates/adherent/terminate_membership.html.twig]: extra spaces: `{attr: {class: 'form__textbox'}}`

File: templates/admin/report/_author.html.twig [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{% set author = object.author %}
+
+{% if author %}
+    <a href="{{ path('admin_app_adherent_show', {'id': author.id}) }}">
```
Comment (Remg) [inline templates/admin/report/_author.html.twig]: useless quotes: `{id: author.id}`

File: templates/report/report.html.twig [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+            <header>
+                <div class="l__wrapper--narrow">
+                    <h1 class="text--center text--larger">
+                        Formulaire de signalement
+                    </h1>
+                </div>
+            </header>
+
+            <section class="l__wrapper--600">
+                {{ form_start(form, { attr: { class: 'b__nudge--top-40'} }) }}
```
Comment (Remg) [inline templates/report/report.html.twig]: extra spaces: `{attr: {class: 'b__nudge--top-40'}}`

File: templates/report/report.html.twig [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+
+                    <div class="form l__col">
+                        <div>
+                            {{ form_row(form.reasons) }}
+                        </div>
+
+                        <div>
+                            {{ form_errors(form.comment) }}
+                            {{ form_widget(form.comment, { attr: { class: 'form__textbox'} }) }}
```
Comment (Remg) [inline templates/report/report.html.twig]: extra spaces: `{attr: {class: 'form__textbox'}}`

**Pull request merged**